### ISSUE TITLE
Aer 0.7.4 Pulse Simulator is incompatible with numpy <1.20.0

### ISSUE THREAD

**Issue by mtreinish** [Qiskit/qiskit-aer#1120]
### Informations

- **Qiskit Aer version**: 0.7.4
- **Python version**: 3.7.9
- **Operating system**: any

### What is the current behavior?

```
Traceback (most recent call last):
  File "/opt/hostedtoolcache/Python/3.7.9/x64/lib/python3.7/unittest/loader.py", line 436, in _find_test_path 
    module = self._get_module_from_name(name)
  File "/opt/hostedtoolcache/Python/3.7.9/x64/lib/python3.7/unittest/loader.py", line 377, in _get_module_from_name
    __import__(name)
  File "/home/vsts/work/1/s/test/python/circuit/test_parameters.py", line 35, in <module>
    from qiskit.test.mock import FakeOurense
  File "/home/vsts/work/1/s/qiskit/test/mock/__init__.py", line 23, in <module>
    from .fake_provider import FakeProvider
  File "/home/vsts/work/1/s/qiskit/test/mock/fake_provider.py", line 22, in <module>
    from .backends import *
  File "/home/vsts/work/1/s/qiskit/test/mock/backends/__init__.py", line 18, in <module>
    from .armonk import FakeArmonk
  File "/home/vsts/work/1/s/qiskit/test/mock/backends/armonk/__init__.py", line 15, in <module>
    from .fake_armonk import FakeArmonk
  File "/home/vsts/work/1/s/qiskit/test/mock/backends/armonk/fake_armonk.py", line 18, in <module>
    from qiskit.test.mock.fake_pulse_backend import FakePulseBackend
  File "/home/vsts/work/1/s/qiskit/test/mock/fake_pulse_backend.py", line 18, in <module>
    from qiskit.test.mock.fake_qasm_backend import FakeQasmBackend
  File "/home/vsts/work/1/s/qiskit/test/mock/fake_qasm_backend.py", line 21, in <module>
    from qiskit.test.mock.fake_backend import FakeBackend
  File "/home/vsts/work/1/s/qiskit/test/mock/fake_backend.py", line 26, in <module>
    from qiskit.providers.aer import Aer
  File "/home/vsts/work/1/s/test-job/lib/python3.7/site-packages/qiskit/providers/aer/__init__.py", line 64, in <module>
    from .aerprovider import AerProvider
  File "/home/vsts/work/1/s/test-job/lib/python3.7/site-packages/qiskit/providers/aer/aerprovider.py", line 20, in <module>
    from .backends.qasm_simulator import QasmSimulator
  File "/home/vsts/work/1/s/test-job/lib/python3.7/site-packages/qiskit/providers/aer/backends/__init__.py", line 20, in <module>
    from .pulse_simulator import PulseSimulator
  File "/home/vsts/work/1/s/test-job/lib/python3.7/site-packages/qiskit/providers/aer/backends/pulse_simulator.py", line 26, in <module>
    from ..pulse.controllers.pulse_controller import pulse_controller
  File "/home/vsts/work/1/s/test-job/lib/python3.7/site-packages/qiskit/providers/aer/pulse/__init__.py", line 52, in <module>
    from .system_models.duffing_model_generators import duffing_system_model
  File "/home/vsts/work/1/s/test-job/lib/python3.7/site-packages/qiskit/providers/aer/pulse/system_models/duffing_model_generators.py", line 21, in <module>
    from .hamiltonian_model import HamiltonianModel
  File "/home/vsts/work/1/s/test-job/lib/python3.7/site-packages/qiskit/providers/aer/pulse/system_models/hamiltonian_model.py", line 22, in <module>
    from .string_model_parser.string_model_parser import HamiltonianParser
  File "/home/vsts/work/1/s/test-job/lib/python3.7/site-packages/qiskit/providers/aer/pulse/system_models/string_model_parser/string_model_parser.py", line 23, in <module>
    from .qobj_from_string import gen_oper
  File "/home/vsts/work/1/s/test-job/lib/python3.7/site-packages/qiskit/providers/aer/pulse/system_models/string_model_parser/qobj_from_string.py", line 18, in <module>
    from ...qutip_extra_lite import qobj_generators
  File "/home/vsts/work/1/s/test-job/lib/python3.7/site-packages/qiskit/providers/aer/pulse/qutip_extra_lite/qobj_generators.py", line 19, in <module>
    from . import operators as ops
  File "/home/vsts/work/1/s/test-job/lib/python3.7/site-packages/qiskit/providers/aer/pulse/qutip_extra_lite/operators.py", line 56, in <module>
    from .fastsparse import fast_csr_matrix, fast_identity
  File "/home/vsts/work/1/s/test-job/lib/python3.7/site-packages/qiskit/providers/aer/pulse/qutip_extra_lite/fastsparse.py", line 440, in <module>
    from .cy.spmath import (zcsr_transpose, zcsr_adjoint, zcsr_mult)
  File "spmath.pyx", line 1, in init qiskit.providers.aer.pulse.qutip_extra_lite.cy.spmath
ValueError: numpy.ndarray size changed, may indicate binary incompatibility. Expected 88 from C header, got 80 from PyObject 
```

### Steps to reproduce the problem

Import the pulse simulator with aer from release on a numpy version < 1.20.0

### What is the expected behavior?

Aer works

### Suggested solutions

Fix the numpy compatibility. My assumption is that this will be fixed in aer 0.8 since the pulse simulator cython code has been removed from master already. That being said we might need to make sure to build with an old numpy version to avoid binary incompatibility if this becomes an issue moving forward.

**Comment by UlysseCambier** [comment]
Thanks for your tip !

**Comment by mtreinish** [comment]
This was fixed in #1131 which was released as 0.7.5 so closing this as fixed, if there are still incompatibilities with 0.7.5 and numpy versions we can open a separate issue on those.
